test/gtest/uct/uct_test.cc

```diff
@@ -239,26 +275,70 @@ void uct_test::set_sockaddr_resources(const md_resource& md_rsc, uct_md_h md,
     for (ifa = ifaddr; ifa != NULL; ifa = ifa->ifa_next) {
         sock_addr.addr = ifa->ifa_addr;
 
-        if (!ucs_netif_flags_is_active(ifa->ifa_flags)) {
-            continue;
-        }
-
-        /* If rdmacm is tested, make sure that this is an IPoIB or RoCE interface */
-        if (!strcmp(md_rsc.rsc_desc.md_name, "rdmacm") &&
-            !ucs::is_rdmacm_netdev(ifa->ifa_name)) {
+        if (!uct_test::is_interface_usable(ifa, md_rsc.rsc_desc.md_name)) {
             continue;
         }
 
         if (uct_md_is_sockaddr_accessible(md, &sock_addr, UCT_SOCKADDR_ACC_LOCAL) &&
             uct_md_is_sockaddr_accessible(md, &sock_addr, UCT_SOCKADDR_ACC_REMOTE))
         {
-            uct_test::set_interface_rscs(md_rsc, local_cpus, ifa, all_resources);
+            uct_test::set_interface_rscs(md_rsc.cmpt, md_rsc.rsc_desc.md_name,
+                                         local_cpus, ifa, all_resources);
+        }
+    }
+
+    freeifaddrs(ifaddr);
+}
+
+void uct_test::set_cm_sockaddr_resources(uct_component_h cmpt, const char *cmpt_name,
+                                         ucs_cpu_set_t local_cpus,
+                                         std::vector<resource>& all_resources) {
+
+    struct ifaddrs *ifaddr, *ifa;
+
+    EXPECT_TRUE(getifaddrs(&ifaddr) != -1);
```

Comment:
can use EXPECT_NE but better check for valid value
also would be useful to print errno in case of error
```
EXPECT_EQ(0, getifaddrs(&ifaddr)) << "errno: " << errno;
```